Clarify security requirements and properties (#178)
tfpauly/privacy-proxy: draft-privacypass-rate-limit-tokens.md

## Information Disclosure {#info-disclosure}

The protocol in this document is designed such that information pertaining to issuance
of a token is limited to parties that need it for completing the protocol. In particular,
Attesters learn only the Anonymous Issuer Origin ID as described in {{anon-issuer-origin-id}},
any per-Client information necessary for attestation, and the target Issuer for a given
token request. The Attester does not learn the origin name associated with a given token
request.

The Issuer only learns the Attester that vouches for a particular Client's token request
and the origin name associated with a token request. The Issuer does not learn the
Anonymous Issuer Origin ID or any per-Client information used when creating a token request.

The Client learns the output token. It does not learn the Anonymous Issuer Origin ID.

## Token Request Unlinkability and Unforgeability {#issuance-unlinkability}

Client token requests are constructed such that an Issuer cannot distinguish between
any two token requests from the same Client and two requests from different Clients.
We refer to this property as issuance unlinkability. This property is achieved

> by construction. In particular, TokenRequest.request_key and TokenRequest.request_signature

Reviewer: ```suggestion
by the way the tokens are constructed. In particular, TokenRequest.request_key and TokenRequest.request_signature
```

are the only value in a TokenRequest that is derived from per-Client information, i.e.,
the Client Secret.

TokenRequest.request_key is computed using a freshly generated blind for each token
request. As a result, the value of TokenRequest.request_key in one token request is
statistically independent from Client Key. Similarly, TokenRequest.request_signature
is computed using the same freshly generated blind as TokenRequest.request_key for each
token request, and the resulting signature is therefore independent from signatures
produced using Client Secret. More details about this unlinkability property can be
found in {{KEYBLINDING}}.

This unlinkability property is only intended for requests observed by the Issuer.
In contrast, the Attester is required to link requests from the same Client together
for the purposes of enforcing rate limits. This Attester does this by observing the
Client Key. Importantly, the Client Key is not sent to the Issuer during the issuance
flow, as doing this would allow the Issuer to trivially link two requests to the same
Client.

The token request signature is also required to be unforgeable. Informally, unforgeability

> means that one cannot produce a valid (message, signature) pair for any blinding key without

Reviewer: ```suggestion
means that no entity can produce a valid (message, signature) pair for any blinding key without
```
